### DISCUSSION TITLE
Deno URL imports in VITE frontend code (not just vite.config.mjs)

### DISCUSSION THREAD

**Discussion by ghost** [denoland/deno#16971] [Ideas]
I would like to work on what's needed to make Deno URL imports work inside a Vite frontend project (TS files in particular).

I'm successfully using Deno + Vite with a Vue TS project but the Deno imports work only inside the `vite.config.mjs`, and also only for the purpose of exposing packages installed through the NPM layer (like `vue` and `vue-router` as shown in [https://deno.com/blog/frameworks-with-npm#vue](https://deno.com/blog/frameworks-with-npm#vue)).

Right now I'm looking into rollup's plugin system as I think one might be needed, but not sure how to connect Deno import system with it and how to make typings work.

Any guidance or suggestion about where to start would very appreciated :v:

**Comment by bartlomieju** [comment]
This is almost the same problem as https://github.com/denoland/deno/issues/17043 - I suggest to continue there. The steps to proceed would be similar to what's described in the issue - shell out to `deno info` to obtain the location of the cached file and return that to Vite's bundler for further processing.

**Comment by ghost** [comment]
Thanks for pointing me to that thread! I'm going to watch that one and comment with any progress I make on my end.

I found out that the goal involves 2 steps:

- Get the rollup plugin working, which indeed can just use deno info and be in charge of the resolution for dev runtime and build.
- The other one is getting this resolution to work inside the IDE (vs code in my case), so that type declarations can also be resolved and so on (this one i'm still investigating if one of the other extensions can support this and how).
